pycloudlib/instance.py

```diff
@@ -41,6 +41,15 @@ def __init__(self, key_pair, username: Optional[str] = None):
         self.connect_timeout = 60
         self.banner_timeout = 60
 
+    def __enter__(self):
+        """Enter context manager for this class."""
+        return self
+
+    def __exit__(self, _type, _value, _traceback):
+        """Exit context manager for this class."""
+        exceptions = self.delete()
+        print_exception_list(exceptions)
```

Comment:
Feels like we do not want a sys.exit(1) on each instance deletion in the cloud, but probably something we need to track when `cloud.clean` is called though.

Reply:
Track how?

Reply:
Sorry. 'track' is a bad word. I want to be noisy if we fail on instance or image deletion or disk deletion as those are costly things to keep around on some clouds.  The [sys.exit(1) was the intent of my comment above to "track" by raising an error via non-zero exit in pycloudlib/cloud.py on __exit__ if teardown/deletion of some resources failed at the end of a test suite](https://github.com/canonical/pycloudlib/pull/305/files#r1248110230).